qcdb/parsers/parse.py

```diff
@@ -50,10 +57,78 @@ def get_library_read_type(self, library_read_type):
         if library_read_type == '2':
             return 'paired-end reverse'
         if library_read_type.isalpha():
-            return 'single ended' 
+            return 'single ended'
         else:
             raise KeyError("Library read type not recognized.")
 
+    # get the reference map for this qc_program (e.g. {kmercount: {long_name: l}})
+    def get_reference_map(self):
+        map = {}
+        s = select([self.ref_table.c.qc_metric, self.ref_table.c.field_name, self.ref_table.c.field_code]).where(self.ref_table.c.qc_program == self.qc_program)
+        for row in self.session.execute(s):
+            # add metric to map if it doesn't yet exist
+            if row['qc_metric'] not in map:
+                map[row['qc_metric']] = {}
+
+            map[row['qc_metric']][row['field_name']] = row['field_code']
+
+        return map
+
+    # utility function to create field code
+    def get_new_mapping(self, field_name, codes):
+        print("Creating new code for ", field_name, " these codes already exist:", codes)
```

Comment:
please use the logging module instead of print statements.

Reply:
all set